src/main/java/org/kumoricon/registration/utility/badgeexport/BadgeExportService.java

```diff
@@ -0,0 +1,241 @@
+package org.kumoricon.registration.utility.badgeexport;
+
+import org.kumoricon.registration.controlleradvice.PrinterSettings;
+import org.kumoricon.registration.guest.Guest;
+import org.kumoricon.registration.guest.GuestRepository;
+import org.kumoricon.registration.model.attendee.Attendee;
+import org.kumoricon.registration.model.attendee.AttendeeRepository;
+import org.kumoricon.registration.model.staff.Staff;
+import org.kumoricon.registration.model.staff.StaffRepository;
+import org.kumoricon.registration.print.BadgePrintService;
+import org.kumoricon.registration.print.Sides;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.*;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.stream.Stream;
+
+@Component
+public class BadgeExportService {
+    private static final Logger log = LoggerFactory.getLogger(BadgeExportService.class);
+
+    private final AttendeeRepository attendeeRepository;
+    private final GuestRepository guestRepository;
+    private final StaffRepository staffRepository;
+    private final BadgePrintService badgePrintService;
+    private final BadgeExport badgeExport;
+
+    public BadgeExportService(AttendeeRepository attendeeRepository, GuestRepository guestRepository, StaffRepository staffRepository,
+                              BadgePrintService badgePrintService, BadgeExport badgeExport) {
+        this.attendeeRepository = attendeeRepository;
+        this.guestRepository = guestRepository;
+        this.staffRepository = staffRepository;
+        this.badgePrintService = badgePrintService;
+        this.badgeExport = badgeExport;
+    }
+
+    public void delegateBadgeExport(PrinterSettings printerSettings) {
+        switch (badgeExport.getType()) {
+            case "Attendee", "Vip", "Specialty" -> exportAttendeeBadgesByType(badgeExport.getPath(), badgeExport.getType(),
+                    badgeExport.isMarkPreprinted(), badgeExport.isWithAttendeeBackground(), printerSettings);
+            case "Staff" -> exportStaffBadges(badgeExport.getPath(), printerSettings);
+            case "Guest" -> exportGuestBadges(badgeExport.getPath(), printerSettings);
+            default -> log.error("{} badge type not recognized", badgeExport.getType());
+        }
+    }
+
+    public void exportAttendeeBadgesByType(String dir, String attendeeType, boolean markPreprinted,
+                                           boolean withAttendeeBackground, PrinterSettings printerSettings) {
+        List<CompletableFuture<String>> futures = new ArrayList<>();
+        List<Attendee> attendees = getAttendeeListByType(attendeeType);
+        if (markPreprinted) {
+            setAttendeePreprinted(attendees);
```

Comment:
this should probably be moved to either after futures are finished being completed or during the actual badge print/generation.. marking them all preprinted here could be inaccurate if futures fail for some reason

Reply:
changed to end of method